source/AsteroidField.cpp

```diff
@@ -125,9 +125,9 @@ void AsteroidField::Draw(DrawList &draw, const Point &center, double zoom) const
 
 
 // Check if the given projectile collides with any asteroids. This excludes minables.
-const vector<Collision> &AsteroidField::CollideAsteroids(const Projectile &projectile) const
+const vector<Collision> AsteroidField::CollideAsteroids(const Projectile &projectile) const
 {
-	result.clear();
+	vector<Collision> result;
```

Comment:
why is this not redefined every time?
beforehand we avoided a lot of memalloc calls

Reply:
It's being accessed concurrently, so we can't store it in the instance

Reply:
ah, can we do std::vector<std::vector<Collision>>(std:thread::concurrent... ?

Reply:
Nah, I don't think the standard library gives enough guarantees for that... not entirely sure, though. I know it allows higher concurrency than the number of cpu threads, but I'm not sure if it allows switching std threads mid-task... probably not.

This solution doesn't actually add a lot of copies, btw, and collision objects are fairly trivial. (Just how many asteroids can a single projectile collide with in a frame?)

Reply:
I mean it is not the size of the vector but how often it gets created, but ig it is fine as is

Reply:
so just to make clear what I mean, if we have a single vector that gets used again, we only have one call to memalloc(expansive cause call to the os), this gets called only once because a clear() will not erase the space, but mark it as unused. memallocs are also not possible to be run in parralel which makes them even more annoying.

But again probably fine as is

Reply:
Honestly, we could probably just have a fixed size collision array `Collision results[10]` or something and it would be fine, without any heap memory.

Yeah, using a single vector would probably be faster, I just wasn't sure if it was worth the trouble (and I haven't really used thread local stuff in C++)

Reply:
thread_local would probably not help given that the thread ends after the loop is left